src/overlays/actors/ovl_En_Gakufu/z_en_gakufu.c

```diff
@@ -34,28 +39,253 @@ const ActorInit En_Gakufu_InitVars = {
+    AudioOcarina_SetInstrumentId(1);
+    AudioOcarina_StartDefault((1 << this->songIndex) | 0x80000000);
+    playbackStaff = AudioOcarina_GetPlaybackStaff();
+    playbackStaff->pos = 0;
+    playbackStaff->state = 0xFF;
+    AudioOcarina_SetInstrumentId(0);
+
+    songIndex = this->songIndex;
+    ocarinaSongButtons = &gOcarinaSongButtons[songIndex];
+    songNumButtons = gOcarinaSongButtons[this->songIndex].numButtons;
+
+    for (i = 0; i < songNumButtons; i++) {
+        this->buttonIdx[i] = ocarinaSongButtons->buttonIdx[i];
+    }
+
+    for (; i < ARRAY_COUNT(this->buttonIdx); i++) {
+        this->buttonIdx[i] = OCARINA_BTN_INVALID;
+    }
+}
+
+void EnGakufu_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnGakufu* this = THIS;
+
+    this->songIndex = OCARINA_SONG_TERMINA_WALL;
+    EnGakufu_ProcessNotes(this);
+    Actor_SetScale(&this->actor, 1.0f);
+
+    if (GAKUFU_GET_TYPE(&this->actor) == GAKUFU_MILK_BAR) {
+        this->actor.draw = NULL;
+        this->actionFunc = EnGakufu_WaitForTimer;
+        return;
+    }
+
+    this->actor.flags &= ~0x2000000;
+
+    if (EnGakufu_IsWithinRange(this, globalCtx)) {
+        gSaveContext.eventInf[3] |= 2;
+    } else {
+        gSaveContext.eventInf[3] &= (u8)~2;
+    }
+
+    this->actionFunc = EnGakufu_WaitForSong;
+    gSaveContext.eventInf[3] &= (u8)~4;
+}
+
+void EnGakufu_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnGakufu* this = THIS;
+
+    if (GAKUFU_GET_TYPE(&this->actor) != GAKUFU_MILK_BAR) {
+        gSaveContext.eventInf[3] &= (u8)~2;
+    }
+}
+
+/**
+ * Special behaviour for the Milk Bar:
+ * this->actor.home.rot.x acts as a timer. Draws the notes on the back of the stage
+ * until the timer runs out
+ */
+void EnGakufu_DisplayOnTimer(EnGakufu* this, GlobalContext* globalCtx) {
+    if (this->actor.home.rot.x > 0) {
+        this->actor.draw = EnGakufu_Draw;
+        this->actor.home.rot.x--;
+    } else {
+        this->actor.draw = NULL;
+        this->actor.home.rot.x = 0;
+        this->actionFunc = EnGakufu_WaitForTimer;
+    }
+}
+
+/**
+ * Special behaviour for the Milk Bar:
+ * this->actor.home.rot.x acts as a timer. Waits for this timer to be set,
+ * then rerolls the notes and start displaying the notes
+ */
+void EnGakufu_WaitForTimer(EnGakufu* this, GlobalContext* globalCtx) {
+    if (this->actor.home.rot.x > 0) {
+        EnGakufu_ProcessNotes(this);
+        this->actionFunc = EnGakufu_DisplayOnTimer;
+    }
+}
+
+void EnGakufu_DoNothing(EnGakufu* this, GlobalContext* globalCtx) {
+}
+
+s32 EnGakufu_IsWithinRange(EnGakufu* this, GlobalContext* globalCtx) {
+    if (this->actor.xzDistToPlayer < 600.0f) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+/**
+ * Reward the player with three item drops depending on the time of day
+ */
+void EnGakufu_GiveReward(EnGakufu* this, GlobalContext* globalCtx) {
+    s32 index;
+    s32 i;
+
+    play_sound(NA_SE_SY_CORRECT_CHIME);
+
+    // 24 hours / The total time in a day
```

Comment:
I feel like this isn't quite what this should say, although we probably currently don't have an appropriate conversion macro for this specific case.